### PR TITLE
HPCC-21087 Conversion of Directories XSD to new format

### PR DESCRIPTION
**Pull request by kenrowland**
Convert XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by kenrowland** [comment]
@rpastrana Please review. This is just the initial conversion to the new format. A second round will review it for correctness.

File: initfiles/componentfiles/configschema/xsd/directories.xsd [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+-->
+
+<xs:schema
+        xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified"
+        xmlns:hpcc="someuri">
+    <xs:include schemaLocation="types.xsd"/>
+    <hpcc:insert hpcc:schemaPath="/Environment/Software">
+        <!-- is a directories section required in an environment? -->
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/directories.xsd]: good question, have you found any insight on who consumes this entry?

File: initfiles/componentfiles/configschema/xsd/directories.xsd [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        <xs:element name="Directories" hpcc:displayName="Directories" hpcc:itemType="directories" hpcc:class="component">
+            <xs:complexType>
+                <xs:sequence>
+                    <xs:element name="Category" hpcc:displayName="Common" maxOccurs="unbounded" type="directorycategory"
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/directories.xsd]: confirm "Common" is the correct display name for this entry

**Review by rpastrana — CHANGES_REQUESTED** [review]
@kenrowland couple of minor questions. Otherwise it seems reasonable (I did not compare to existing XSD)

File: initfiles/componentfiles/configschema/xsd/directories.xsd (hunk 1 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/directories.xsd]: No. The comment is there for when it goes to the component owner.

File: initfiles/componentfiles/configschema/xsd/directories.xsd (hunk 2 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/directories.xsd]: Component owner to confirm this level of correctness. I only translated it from the current XSD.

**Review by kenrowland — COMMENTED** [review]
@rpastrana  Please see responses.

File: initfiles/componentfiles/configschema/xsd/directories.xsd (hunk 1 above)
Comment (rpastrana) [reply initfiles/componentfiles/configschema/xsd/directories.xsd]: who is the component owner?

**Review by rpastrana — APPROVED** [review]
@kenrowland

File: initfiles/componentfiles/configschema/xsd/directories.xsd (hunk 1 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/directories.xsd]: @rpastrana Jake, I think. The owner will get a JIRA to review the XSD for correctness.

**Comment by kenrowland** [comment]
@richardkchapman Please merge

**Pull request merged**